crates/polars-io/src/csv/read/splitfields.rs

```
@@ -369,14 +376,45 @@ mod inner {
                         let has_separator = simd_bytes.simd_eq(self.simd_separator);
                         let has_any_mask = (has_separator | has_eol_char).to_bitmask();
 
+                        // check malformed: quotes are not allowed in unenclosed fields
+                        if self.quoting {
+                            let has_quote_mask =
+                                simd_bytes.simd_eq(self.simd_quote_char).to_bitmask();
```

I don't think it's worth taking the performance hit of this extra code in the hot-loop for the malformed quote case, especially as it affects the performance in benchmarks.

I think the best strategy here is to just "let it fail" - i.e. proceed to parsing as usual and allow it to fail.

With regards to the original issue at https://github.com/pola-rs/polars/pull/22876, I believe the key requirement on our side is that we behave consistently (i.e. return the same result for `read_csv` / `scan_csv` / `scan_csv().select(nth(0))`).